### DISCUSSION TITLE
<opensafely upgrade> error in Anaconda prompt

### DISCUSSION THREAD

**Discussion by ciaranmci** [opensafely/documentation#568] [Installing OpenSAFELY] [answered]
Every time I submit command in my Anaconda prompt I'm told that v1.17.1 of opensafely is available (I currently have 1.15.1). But everytime I run <opednsafely upgrade>, it says it cannot find a version that satisfies the requirement for opensafely =1.17.1 (despite listing 1.15.1 in the list of options).

Has anyone else experienced this issue?

**Comment by StevenMaude** [comment, marked as answer]
@ciaranmci: can you check what version of Python you have? `python --version` within Anaconda Prompt should show you.

The latest version of the opensafely-cli [recently started requiring Python 3.8](https://github.com/opensafely-core/opensafely-cli/commit/8798730786368e343e44b67853fa822fc12aee9a), so this might be an issue here if you are on Python 3.7.

**Reply by ciaranmci** [reply]
Thanks for replying, Steven.
I've finally upgraded to Python3.10.2 but the <opensafely upgrade> command still returns the same error (although now it seems that opensafely v1.18.0 is available!).

Any other ideas? Shall I just uninstall and reinstall?

So far, this issues hasn't stopped me running anything locally. I'm still in the co-pilot stage so I will soon test everything on the proper server

**Reply by StevenMaude** [reply]
@ciaranmci: Sorry for the slow reply; I lost this in my GitHub notifications!

I don't use Anaconda, so someone else reading may be able to provide more specific advice. But, if you haven't already, I'd be roughly trying:

1. Uninstall `opensafely-cli` entirely
2. Confirm that you're in an Anaconda Prompt running Python 3.10
3. Install `opensafely-cli` again

**Reply by ciaranmci** [reply]
No luck, I'm afraid.
It keeps installing openasafely v1.15.1, even when I try to force v1.20.0 (pip install opensafely==1.20.0)
![opensafely-cli error](https://user-images.githubusercontent.com/44294573/153389571-2924b92a-2a9d-4833-86a3-2acbbbc8961a.JPG)

**Reply by StevenMaude** [reply]
:arrow_up: It looks like that `pip install` is still running with an old Python.

It seems like you might need to create a new environment with the new version of Python, maybe? I'm not sure of the current canonical way of doing that. If someone else is reading who does know, they might give more concrete advice.

Things I found from a quick look around:

* [Managing Python versions with Anaconda Navigator](https://docs.anaconda.com/anaconda/navigator/tutorials/use-multiple-python-versions/)
* [Managing Python versions with `conda`](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-python.html)

If you can Anaconda Prompt to use a Python 3.8 or later, then the `pip install opensafely==1.20.0` should work.

**Reply by Jongmassey** [reply]
if you want to install python 3.10 in your base conda image then ` conda install python=3.10` will do the trick. If you wish to keep your base environment and on a different version then `conda create --name my_shiny_new_env`, `conda activate my_shiny_new_env` and ` conda install python=3.10` will install the latest python in this isolated environment leaving your base environment intact

**Reply by ciaranmci** [reply]
Thanks @Jongmassey and @StevenMaude.
University IT finally approved the conda install of py 3.10 and it permitted me to install OpenSAFELY 1.20.1
Really appreciate your help.
Happy to close this issue.

**Reply by StevenMaude** [reply]
@ciaranmci: thanks for letting us know you got it fixed, and glad we could help! :partying_face:
